lib/LongVectorLoweringPass.cpp

```diff
@@ -270,20 +270,38 @@ Function *getBIFScalarVersion(Function &Builtin) {
 
   // TODO Add support for other builtins by providing testcases and listing the
   // builtins here.
+  case clspv::Builtins::kExp:
+  case clspv::Builtins::kFma:
+  case clspv::Builtins::kFmax:
+  case clspv::Builtins::kFmin:
+  case clspv::Builtins::kFrexp:
   case clspv::Builtins::kMax: {
     // Scalarise all the input/output types. Here we intentionally do not rely
     // on getEquivalentType because we want the scalar overload.
-    SmallVector<Type *, 16> ParamTys;
+    SmallVector<Type *, 16> ScalarParamTys;
     for (auto &Param : Builtin.args()) {
-      // TODO Need support for other types, like pointers. Need test case.
-      assert(Param.getType()->isVectorTy());
-      ParamTys.push_back(Param.getType()->getScalarType());
+      auto *ParamTy = Param.getType();
+
+      Type *ScalarParamTy = nullptr;
+      if (ParamTy->isVectorTy()) {
```

Comment:
I missed this for previously, but several of these (e.g. fmax) the second argument maybe be a scalar to be splatted. I would suggest flipping this around to test if the parameter is a pointer so that you can grab the right type in such situations. This will avoid needing the splat arg pass earlier in the flow.
```
if (ParamTy->isPointerTy()) {
  ...
} else {
  ScalarParamTy = ParamTy->getScalarTy();
}
```

This deserves a test.

Reply:
Good catch. I've updated the code & test to handle this.
(PS: I'll be away for a bit but if there is any other comments I'll address them when I'm back early next year.)